tests/test_cli.py

```diff
@@ -0,0 +1,213 @@
+def test_browser_argument(browser_arg):
+    """Test arguments for the browser option."""
+    for browser_opt in VALID_CMD_OPTS[2]:
+        output = subprocess.Popen(
+            [CMD_ROOT, browser_opt, browser_arg],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout, stderr = output.communicate()
+        if output.returncode == 0:
+            assert CSV_HISTORY_HEADER in stdout.decode("utf-8")
+        elif any(
+            browser_unavailable_err in stderr.decode("utf-8")
+            for browser_unavailable_err in (
+                "browser is not supported",
+                "browser is not installed",
+            )
+        ):
+            # In case the tester does not have access to the browser
+            # in question, which makes the command fail, but in a way
+            # that is expected and gives a recognised error message.
+            pytest.skip(
+                "Unable to test against this browser because it is "
+                "not available locally"
+            )
+        else:  # command fails for another reason i.e. a test failure
+            pytest.fail(
+                "The browser is available but the command to fetch "
+                "the history from it has failed."
+            )
+
+
+def test_format_argument():
+    """Tests arguments for the format option."""
+    # First check format of default:
+    csv_output = subprocess.check_output([CMD_ROOT]).decode("utf-8")
+    # Sniffer determines format, less intensive than reading in csv.reader
+    # and we don't mind the CSV dialect, so just check call doesn't error
+    read_csv = csv.Sniffer()
+    read_csv.sniff(csv_output, delimiters=",")
+    read_csv.has_header(csv_output)
+
+    for fmt_opt in VALID_CMD_OPTS[3]:
+        for fmt_arg in VALID_FORMAT_ARGS:
+            output = subprocess.check_output([CMD_ROOT, fmt_opt, fmt_arg]).decode(
+                "utf-8"
+            )
+            if fmt_arg in ("csv", "infer"):  # infer gives csv if no file
+                csv.Sniffer().sniff(output, delimiters=",")
+                csv.Sniffer().has_header(output)
+                assert CSV_HISTORY_HEADER in output
+            elif fmt_arg == "json":
+                assert output.startswith(HISTORY_TITLE)
+                # Need to strip header off before checking for valid json
+                json.loads(output.lstrip(HISTORY_TITLE))
+            else:  # i.e. fmt_arg == 'jsonl':
+                assert output.startswith(HISTORY_TITLE)
+                jsonl_output = output.lstrip(HISTORY_TITLE)
+                # Newline-delimited json so test each line is valid json
+                for line in jsonl_output.splitlines():
+                    json.loads(line)
+
+
+def test_output_argument():
+    """Test arguments for the output option."""
+    for output_opt in VALID_CMD_OPTS[4]:
+        with tempfile.NamedTemporaryFile(suffix=".csv", delete=False) as f:
```

Comment:
Is the file used for something else? Or is there a different reason for keeping the temp file?

Reply:
At the time I thought `delete=False` might be required for the file to persist so the sub-process will see it, but I've just tried the tests with that specified with it removed, and they still work. So it isn't needed after all. I'll remove both cases in the next commit. Thanks.